File: src/main/java/dev/zprestige/fire/module/movement/blink/TickListener.java
package dev.zprestige.fire.module.movement.blink;

import dev.zprestige.fire.newbus.EventListener;
import dev.zprestige.fire.newbus.events.TickEvent;

public class TickListener extends EventListener<TickEvent, Blink> {

    public TickListener(final Blink blink) {
        super(TickEvent.class, blink);
    }

    @Override
    public void invoke(final Object object) {
        if (module.mode.GetCombo().equals("Pulse")) {
            module.i++;
        }
        if (module.i >= module.ticks.GetSlider()) {
            module.poll(true);
            module.spawnEntity();
            module.i = 0;
        }
    }
}

File: src/main/java/dev/zprestige/fire/module/movement/blink/TickListener.kt
package dev.zprestige.fire.module.movement.blink

import dev.zprestige.fire.event.bus.EventListener
import dev.zprestige.fire.event.impl.TickEvent

class TickListener(blink: Blink) : EventListener<TickEvent, Blink>(TickEvent::class.java, blink) {
    
    override fun invoke(`object`: Any) {
        if (module.mode.GetCombo() == "Pulse") {
            module.i++
        }
        if (module.i >= module.ticks.GetSlider()) {
            module.poll(true)
            module.spawnEntity()
            module.i = 0
        }
    }
}